Add note about keeping the middleware pipeline running for a WebSocket (#12378)
aspnet/AspNetCore.Docs: aspnetcore/fundamentals/websockets.md

A WebSocket request could come in on any URL, but this sample code only accepts requests for `/ws`.

When using a WebSocket, you **must** keep the middleware pipeline running for the duration of the connection. If you attempt to send or receive a WebSocket message after the middleware pipeline ends, you may get an exception like the following:

```
System.Net.WebSockets.WebSocketException (0x80004005): The remote party closed the WebSocket connection without completing the close handshake. ---> System.ObjectDisposedException: Cannot write to the response body, the response has completed.
Object name: 'HttpResponseStream'.
```

> If you are using a background service to write data to a WebSocket, make sure you keep the middleware pipeline running. You can do this by using a <xref:System.Threading.Tasks.TaskCompletionSource%601>. Pass the TaskCompletionSource to your background service and have it call <xref:System.Threading.Tasks.TaskCompletionSource%601.TrySetResult%2A> when you finish with the WebSocket, then `await` the <xref:System.Threading.Tasks.TaskCompletionSource%601.Task> property during the request.

Reviewer: ```suggestion
If you're using a background service to write data to a WebSocket, make sure you keep the middleware pipeline running. Do this by using a <xref:System.Threading.Tasks.TaskCompletionSource%601>. Pass the `TaskCompletionSource` to your background service and have it call <xref:System.Threading.Tasks.TaskCompletionSource%601.TrySetResult%2A> when you finish with the WebSocket. Then `await` the <xref:System.Threading.Tasks.TaskCompletionSource%601.Task> property during the request.
```

Author: FOILED AGAIN!

Author: Contractions... my old adversary.

![sideshow bob from the simpsons stepping on a lot of rakes](https://media1.tenor.com/images/d99bd8d655f494fc4263cc044ffc309f/tenor.gif?itemid=5683040)
